Frontend/src/app/home/search-and-filter/search-and-filter.component.ts

```diff
@@ -0,0 +1,76 @@
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { FilterDialogComponent } from '../filter-dialog/filter-dialog.component';
+import { MatIcon } from '@angular/material/icon';
+import { MatButton } from '@angular/material/button';
+import { CategoryService } from '../../services/category.service';
+import { PriceCategoryService } from '../../services/price-category.service';
+import { MatDialog } from '@angular/material/dialog';
+import { FilterDialogData } from '../../model/filter-dialog-data.model';
+import { FilterData } from '../../model/filter-data.model';
+import { filter } from 'rxjs';
+
+@Component({
+	selector: 'app-search-and-filter',
+	standalone: true,
+	imports: [MatIcon, MatButton],
+	templateUrl: './search-and-filter.component.html',
+	styleUrl: './search-and-filter.component.scss',
+})
+export class SearchAndFilterComponent implements OnInit {
+	private readonly filterDialogData: FilterDialogData;
+	private readonly filter: FilterData;
+
+	@Output() filterApplied = new EventEmitter<FilterData>();
+
+	constructor(
+		private categoryService: CategoryService,
+		private priceCategoryService: PriceCategoryService,
+		private dialog: MatDialog
+	) {
+		this.filterDialogData = {
+			selectedFoodCategories: [],
+			priceCategories: [],
+			selectedPriceCategories: [],
+			foodCategories: [],
+		};
+
+		this.filter = {
+			text: '',
+			filterDialogData: this.filterDialogData,
+		};
```

Comment:
Can't you make filterData an input property and pass the initialized filterData from the other component that way you don't have to initialize it twice?